src/game/logic/object/object.h

```diff
@@ -0,0 +1,620 @@
+    float Get_Construction_Percent() const { return m_constructionPercent; }
+    AIUpdateInterface *Get_AI_Update_Interface() { return m_ai; }
+    const AIUpdateInterface *Get_AI_Update_Interface() const { return m_ai; }
+    PathfindLayerEnum Get_Layer() const { return m_layer; }
+    const GeometryInfo &Get_Geometry_Info() const { return m_geometryInfo; }
+    Drawable *Get_Drawable() const { return m_drawable; }
+    Utf8String Get_Name() { return m_name; }
+    const PhysicsBehavior *Get_Physics() const { return m_physics; }
+    PhysicsBehavior *Get_Physics() { return m_physics; }
+    const ContainModuleInterface *Get_Contain() { return m_contain; }
+    const Object *Get_Contained_By() const { return m_containedBy; }
+    Object *Get_Contained_By() { return m_containedBy; }
+    Module **Get_All_Modules() { return m_allModules; }
+    ObjectID Get_Producer_ID() const { return m_producerID; }
+    unsigned int Get_Weapon_Bonus_Condition() const { return m_weaponBonusCondition; }
+    const Weapon *Get_Weapon_In_Weapon_Slot(WeaponSlotType type) const
+    {
+        return m_weaponSet.Get_Weapon_In_Weapon_Slot(type);
+    }
+    const ExperienceTracker *Get_Experience_Tracker() const { return m_experienceTracker; }
+    ExperienceTracker *Get_Experience_Tracker() { return m_experienceTracker; }
+    const BitFlags<WEAPONSET_COUNT> &Get_Weapon_Set_Flags() const { return m_curWeaponSetFlags; }
+    const Team *Get_Team() const { return m_team; }
+    Team *Get_Team() { return m_team; }
+    StealthUpdate *Get_Stealth_Update() const { return m_stealth; }
+    bool Get_Script_Status(ObjectScriptStatusBit status) const { return (m_scriptStatus & status) != 0; }
+    Object *Get_Next_Object() { return m_next; }
+    const BitFlags<OBJECT_STATUS_COUNT> &Get_Status_Bits() const { return m_status; }
+    BodyModuleInterface *Get_Body_Module() const { return m_body; }
+    bool Get_Disabled_State(DisabledType type) const { return m_disabledStates.Get(type); }
+    const BitFlags<DISABLED_TYPE_COUNT> &Get_Disabled_State_Bits() const { return m_disabledStates; }
+    bool Get_Receiving_Difficulty_Bonus() { return m_receivingDifficultyBonus; }
+    BitFlags<128> Get_Object_Upgrade_Mask() const { return m_objectUpgradesCompleted; }
+    bool Get_Apply_Battle_Plan_Bonuses() const { return m_applyBattlePlanBonuses; }
+    PathfindLayerEnum Get_Destination_Layer() const { return m_destinationLayer; }
+    void Get_Formation_Offset(Coord2D *c) { *c = m_formationOffset; }
+    bool Get_Single_Use_Command() const { return m_singleUseCommand; }
+
+    void Clear_Status(BitFlags<OBJECT_STATUS_COUNT> bits) { return Set_Status(bits, false); }
+    bool Clear_Script_Status(ObjectScriptStatusBit bit) { return Set_Script_Status(bit, false); }
+
+    Module *Find_Module(NameKeyType type) const;
+    DamageModule *Find_Damage_Module(NameKeyType type) const { return (DamageModule *)Find_Module(type); }
```

Comment:
Don't use a c style cast for this it could be subtly broken if damage module uses multiple inheritance. Use a static cast instead

Reply:
The original code did the equivalent of a C style cast so its perfectly OK for our clone to do the same.

Reply:
I've already had one bug in this project that was due to this just wanting to stop it from happening again. If you can guarantee that DamageModule doesn't / will never use multiple inheritance then yes its fine.